File: src/main/java/com/sfh/pokeRogueBot/model/run/ChosenAttackMove.java
package com.sfh.pokeRogueBot.model.run;

import com.sfh.pokeRogueBot.model.enums.ChoosenAttackMoveType;
import com.sfh.pokeRogueBot.model.enums.MoveTargetAreaType;
import com.sfh.pokeRogueBot.model.enums.OwnPokemonIndex;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class ChosenAttackMove {

    private int index;
    private String name;
    private int damage;
    private ChoosenAttackMoveType choosenAttackMoveType;
    private int attackPriority;
    private int attackerSpeed;
    private OwnPokemonIndex ownPokemonIndex;
    private MoveTargetAreaType moveTargetAreaType;
}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/decisions/ChosenAttackMove.kt
package com.sfh.pokeRogueBot.model.decisions

import com.sfh.pokeRogueBot.model.enums.ChoosenAttackMoveType
import com.sfh.pokeRogueBot.model.enums.MoveTargetAreaType
import com.sfh.pokeRogueBot.model.enums.OwnPokemonIndex

data class ChosenAttackMove(
    val index: Int,
    val name: String,
    val damage: Int,
    val choosenAttackMoveType: ChoosenAttackMoveType,
    val attackPriority: Int,
    val attackerSpeed: Int,
    val ownPokemonIndex: OwnPokemonIndex,
    val moveTargetAreaType: MoveTargetAreaType
)